docs/news/posts/2024-11-15-aiidalab-mvc.md

```diff
@@ -0,0 +1,322 @@
+
+```python
+
+import traitlets as tl
+
+class Model(tl.HasTraits):
+    attr1 = tl.Unicode(...)
+    attr2 = tl.Float(...)
+    attr3 = tl.Bool(...)
+
+    def update(self): ...
+    def reset(self): ...
+```
+
+Trait (state) changes can be observed and acted upon by the `Controller`, updating the `MVC` component (and network) according to the defined logic. The specific use (the API) of traits in the `View` and `Controller` classes will be discussed in their respective sections.
+
+### The `View`
+
+Once the logic of the app is established and tested by leveraging the model network, its time to connect it to a `View` to enable user interactions. In AiiDAlab, views are _mostly_ constructred from UI components provided by the [ipywidgets](https://ipywidgets.readthedocs.io/en/stable/) Python UI library. The library also provides means to construct [custom widgets](https://ipywidgets.readthedocs.io/en/stable/examples/Widget%20Custom.html). Alternatively, one could explore the [anywidget](https://anywidget.dev/) API for developing custom UI components.
+
+A typical view may look like the following:
+
+```python
+
+import ipywidgets as ipw
+
+class View(ipw.VBox):
+
+    def __init__(self, **kwargs):
+        self.dropdown = ipw.Dropdown(description="Options")
+        self.textbox = ipw.Text(description="Label")
+        self.button = ipw.Button(
+            description="Submit",
+            button_style="success",
+            icon="check",
+        )
+        super().__init__(
+            children=[
+                self.dropdown,
+                ipw.HBox(
+                    children=[
+                        self.textbox,
+                        self.button,
+                    ],
+                ),
+            ],
+        )
+```
+
+![View example]()
+
+The class represents a UI for the user to interact with the app. It typically provides an interface for the user to select and provide input for the underlying simulation software.
+
+**Tip 💡** consider arranging UI components in a pattern closely representative of the flow of app operations. This is often best determined hand-in-hand with the intended user base.
+
+#### `View`-`Model` synchronization
+
+The observable state variables defined in the `Model` can be leveraged to synchronize the `View` with the current state of the app. This can be done via the `traitlets` linking mechanism. This can be set up in the `Controller` class.
+
+### The `Controller`
+
+The purpose of the `Controller` class, the final component of the MVC design patter, is to handle events and control the flow of information across the app. This means reacting to changes in models and/or user-driven view events, updating either according to the defined app logic. To do so, a `Controller` is typically instantiated with instances of both the `View` and the `Model`, as follows:
```

Comment:
```suggestion
The purpose of the `Controller` class, the final component of the MVC design pattern, is to handle events and control the flow of information across the app. This means reacting to changes in models and/or user-driven view events, updating according to the defined app logic. To do so, a `Controller` is typically instantiated with instances of both the `View` and the `Model`, as follows:
```

Reply:
Good catch on "patter", but why take "either" out? The `Controller` updates EITHER the model or the view 🤔

Reply:
Ah, OK, now I get it. I read it as "...updating either according to the defined app logic <_or something else_>.", didn't realize the _either_ referred to the sentence **before**. Maybe add a comment afterwards to make it more clear.